### DISCUSSION TITLE
配合chinadns-ng增强dns防污染

### DISCUSSION THREAD

**Discussion by w311ang** [coolsnowwolf/lede#9852] [Show and tell]
gfwlist并不完善，总有漏网之鱼，导致了DNS被污染，基于chinadns的[工作原理](https://github.com/zfl9/chinadns-ng#%E5%B7%A5%E4%BD%9C%E5%8E%9F%E7%90%86)，简单来说就是国内DNS回应的污染IP一定是国外IP，根据这个规律，一旦国内DNS回应了国外IP那就等待可信DNS回应正确的国外IP

编译前，menuconfig选中luci-chinadns-ng
```bash
git clone https://github.com/pexcn/openwrt-chinadns-ng.git package/chinadns-ng
git clone -b luci https://github.com/w311ang/openwrt-chinadns-ng.git package/luci-app-chinadns-ng
```

进入chinadns-ng设置，监听端口5353，可信dns改`127.0.0.1#5335`，把所有节点的域名都添加到白名单

打开DHCP/DNS设置，设置DNS转发`127.0.0.1#5353`，启动项添加并ssh运行`uci set dhcp.@dnsmasq[0].noresolv="1"; uci commit dhcp; /etc/init.d/dnsmasq restart`<sup>[issue](https://github.com/fw876/helloworld/issues/951)</sup>，若使用dnsforwarder并且未配置`addr`命令这一步可以省略

这样就完成了，你会发现之前打不开的一些网站现在可以打开了

**Comment by jliveusa** [comment]
请问这种情况下ssrp的dns怎么设置？使用5335端口的服务吗？谢谢
